src/utils/debounce.ts

```diff
@@ -0,0 +1,19 @@
+export default function (delay: number) {
+  let timeoutId: ReturnType<typeof setTimeout> | undefined
+  let leadingExecuted = false
+
+  return (callback: () => void) => {
+    if (!leadingExecuted) {
+      callback()
+      leadingExecuted = true
+    } else {
+      if (timeoutId) {
+        clearTimeout(timeoutId)
+      }
+      timeoutId = setTimeout(async () => {
+        await callback()
```

Comment:
Should we invoke this call here?
Also, does the callback signature (`callback: () => void`) support await usages?

Reply:
yes, this is the delayed execution
await works with any function AFAIK

Reply:
I see this message in the editor BTW
```
'await' has no effect on the type of this expression
```

Reply:
Also, my reasoning for questioning invocation of callback here is, if we invoke it here and set the `leadingExecuted` to false, the callback also gets executed during the next invocation based on condition here https://github.com/Nosto/nosto-autocomplete/pull/209/files#diff-27deecf0e921d131498e3d2e9b50d61d0444ce5740cdbdd716ab531e6f36e552R6
Is that the intention?

Reply:
* 'await' has no effect on the type of this expression
  * I can change the callback signature to make sure that both sync and async functions can be handled, but is that necessary?
* yes, that's the intention